lms/handle-invalid-ids-in-vitally-unlink

anathomical opened the pull request:
## WHAT
Allow for cases where the Vitally data has invalid user ids
## WHY
Apparently the data file Vitally sent us includes some nonsense data in it which we need to account for
## HOW
Use `find_by` so that we don't error if the `id` value from Vitally is invalid, and use `next` to skip cases where there is no User.

### Notion Card Links
https://www.notion.so/quill/Script-to-unlink-users-who-have-been-associated-with-multiple-accounts-in-Vitally-65df89265b8a44beb05bd205841a7208?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No tests on rake tasks
Have you deployed to Staging? | Tiny change
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

anathomical committed:
Allow for cases where the Vitally data has invalid user ids

brendanshean commented on services/QuillLMS/lib/tasks/vitally.rake:
```diff
@@ -19,7 +19,12 @@ namespace :vitally do
+
+        # Some of the data provided by Vitally is bad, so we need to account for that
+        next unless user
+
```
Very minor, but if we're adding a joins, is safe navigation necessary here:
 `quill_school_id = user.schools_users&.school_id`

anathomical replied:
Good catch.  We actually want cases where there are no school assignments, so I think I actually need to make the `joins` into a `eager_load`.

brendanshean commented on services/QuillLMS/lib/tasks/vitally.rake:
```diff
@@ -19,7 +19,12 @@ namespace :vitally do
     CSV.parse(pipe_data, headers: true).each_slice(33) do |batch|
       batch_payload = batch.map do |row|
-        user = User.includes(:schools_users).find(row['externalId'])
+        # TIL `includes` makes a separate query to the db, and `joins` doesn't store the joined value in ActiveRecord, so if you want a single query to pull multiple models you need to use both?
```
I usually have to look this up.  Reminds me of the other [options](https://www.bigbinary.com/blog/preload-vs-eager-load-vs-joins-vs-includes) preload and eager_load too.  I feel like at one point Rails changed how includes worked which I think is why I find this confusing.

anathomical committed:
Use eager_load instead of joins + includes

It gives us the actual behavior we want: a single query for both records, but gives us a User even if there is no associated SchoolsUser
